presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -316,13 +316,7 @@ public static boolean isRoleEnabled(ConnectorIdentity identity, Function<HivePri
             return false;
         }
 
-        HivePrincipal principal;
-        if (!identity.getRole().isPresent() || identity.getRole().get().getType() == SelectedRole.Type.ALL) {
-            principal = new HivePrincipal(USER, identity.getUser());
-        }
-        else {
-            principal = new HivePrincipal(ROLE, identity.getRole().get().getRole().get());
-        }
+        HivePrincipal principal = HivePrincipal.from(identity);
```

Comment:
is this new method applicable here too?
https://github.com/prestosql/presto/blob/c672138fde86472ab470aaa7943f61e010387007/presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java#L347-L353